### PR TITLE
x32edit,m32edit: 4.1 -> 4.3

### PR DESCRIPTION
**Pull request by ck3d**
Update x32edit and m32edit to the latest version 4.3. The program starts, but please check if all functions work as expected.
Since version 4.3 (built 2021) the executable is linked against libcurl3-gnutls, which is not available in nixpkgs.
I had to link the original Debian package to the executable. The approach is novel, so please have a deeper look.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

cc @Korny666  @markuskowa

### CODE REVIEW

**Review by markuskowa — COMMENTED** [review]
I think this is an elegant solution for a pathological case.

**Comment by Korny666** [comment]
x32 seems to work fine!
m32 i cannot test with hardware

File: pkgs/build-support/vm/default.nix [hunk 1]
```diff
@@ -726,7 +726,7 @@ rec {
     {name, packagesLists, urlPrefix, packages}:
 
     runCommand "${name}.nix"
-      { nativeBuildInputs = [ buildPackages.perl buildPackages.dpkg ]; } ''
```
Comment (matthiasbeyer) [inline pkgs/build-support/vm/default.nix:729]: How did this pass review?  @Korny666 @markuskowa

File: pkgs/build-support/vm/default.nix [hunk 2]
```diff
@@ -744,6 +744,7 @@ rec {
 
       perl -w ${deb/deb-closure.pl} \
         ./Packages ${urlPrefix} ${toString packages} > $out
+      nixfmt $out
```
Comment (matthiasbeyer) [inline pkgs/build-support/vm/default.nix:747]: How did this pass review?  @Korny666 @markuskowa

Comment (Korny666) [reply pkgs/build-support/vm/default.nix:747]: Can you explain a bit further what you are trying to point out?
I am a bit confused 🫠

**Pull request merged**